hooks/checkout

```diff
@@ -139,18 +165,14 @@ copy_checkout_from_s3() {
     aws s3 cp "${s3_url}/${checkout}" "${PWD}/${checkout}"
   fi
 
-  local -a decompress_options
-  if [[ "$(basename "${checkout}" ".zst")" != "${checkout}" ]]; then
-    # Decompress using zstd
-    decompress_options=("-I" "zstd")
-  else
-    # Auto-detect decompression options
-    decompress_options=()
-  fi
-
-  tar "${decompress_options[@]}" -xf "${PWD}/${checkout}"
+  extract_archive "${PWD}/${checkout}"
   rm "${PWD}/${checkout}"
-  popd >/dev/null
+
+  # Check if extracted archive is a repository
+  if [[ "$(git rev-parse --is-inside-work-tree 2>/dev/null)" != "true" || -n "$(git check-ignore .)" ]]; then
+    # Change directory to first child when repository is not found
+    pushd $(ls -d */ | head -n 1)
+  fi
```

Comment:
when would this be required?
I don't think this is going to work since the repo would then be located in a different directory to where buildkite will try to use.